### DISCUSSION TITLE
Background threads in synchronous python-socketio Client

### DISCUSSION THREAD

**Discussion by ramanraja** [miguelgrinberg/python-socketio#1019] [Q&A] [answered]
My question is different from the scenario in `https://github.com/miguelgrinberg/python-socketio/discussions/984`
I have a  python-socketio **Client** program. It communicates with a remote socketio server, which may be synchronous or asynchronous. But my Client is purely a synchronous one, installed by
`pip install python-socketio[client]`
I have a need to spawn a few 'normal' Python threads (not compatible with asyncio) in the background. These threads may be blocking indefinitely. The threads are spawned by a third party library, so I cannot use the start_background_task() utility.
The main program thread will also be parked in a loop with
`sio.wait()`
Is this a technically sound approach? I have a small test implementation of this idea, which seems to work fine. 
But when production volumes of traffic starts, will the background threads interfere with the socket client in any way?

**Comment by miguelgrinberg** [comment, marked as answer]
The style in which the server is written is irrelevant to the client, so that is not something that you need to worry about. The sync or async style matters within the same process only.

If your client uses threads, then you should be fine, since a blocked thread does not block other threads.
